### PR TITLE
feat: add badge component

### PR DESCRIPTION
**Pull request by vtrbo**
- add badge component
- add badge docs

### CODE REVIEW

File: preset/src/shortcuts/badge.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export const badgeShortcuts: Record<string, string> = {
+	// badge
+	'k-badge--base': 'relative v-mid inline-block w-fit',
+	'k-badge--content':
+		'inline-flex justify-center items-center h-18px b-rd-10px text-12px p-y-0 p-x-6px whitespace-nowrap b-1px b-solid b-ikun-white color-ikun-white',
+	'k-badge--content-fixed': 'absolute top-0 right-10px translate-y--50% translate-100% z-1',
```
Comment (baiwusanyu-c) [inline preset/src/shortcuts/badge.ts:6]:
```suggestion
	'k-badge--content-fixed': 'pa top-0 right-10px translate-y--50% translate-100% z-1',
```

File: preset/src/shortcuts/badge.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export const badgeShortcuts: Record<string, string> = {
+	// badge
+	'k-badge--base': 'relative v-mid inline-block w-fit',
+	'k-badge--content':
+		'inline-flex justify-center items-center h-18px b-rd-10px text-12px p-y-0 p-x-6px whitespace-nowrap b-1px b-solid b-ikun-white color-ikun-white',
```
Comment (baiwusanyu-c) [inline preset/src/shortcuts/badge.ts:5]: You can add an `inline-flex justify-center items-center` combination in `shortcuts/common.ts`, just like `infsc`

File: docs/example/badge/max-value.svelte [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+	let count = 12;
+</script>
+
+<KBadge value={count} max={12} cls="mr-40px">
+	<KButton on:click={() => count++}>让我怎么荔枝</KButton>
+</KBadge>
+
+<KBadge value={666} max={25} type="warning" cls="mr-40px">
+	<KButton type="info">小黑子</KButton>
```
Comment (baiwusanyu-c) [inline docs/example/badge/max-value.svelte:13]: example中所有梗用英文代替吧

File: components/Badge/src/index.svelte [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<script lang="ts">
+	import { isNumber } from 'baiwusanyu-utils';
+	import type { IKunTypePro } from '@ikun-ui/utils';
+
+	export let value: string | number = '';
+	export let max: number = 99;
+	export let isDot: boolean = false;
+	export let hidden: boolean = false;
```
Comment (baiwusanyu-c) [inline components/Badge/src/index.svelte:8]: What about `show` instead? Other components do not use `hidden`

**Review by baiwusanyu-c — CHANGES_REQUESTED** [review]

File: docs/components/KBadge.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+<demo src="../../../../example/badge/customizations.svelte" github='https://github.com/ikun-svelte/ikun-ui/tree/main/components/Button'></demo>
+
+## Dot Mark
+
+Use a dot to mark content that needs to be noticed.
+
+Use the attribute is-dot. It is a Boolean.
+
+<demo src="../../../../example/badge/dot-mark.svelte" github='https://github.com/ikun-svelte/ikun-ui/tree/main/components/ButtonGroup'></demo>
```
Comment (baiwusanyu-c) [inline docs/components/KBadge.md:58]: source code link error

**Review by baiwusanyu-c — CHANGES_REQUESTED** [review]

File: docs/example/badge/max-value.svelte (hunk 3 above)
Comment (vtrbo) [reply docs/example/badge/max-value.svelte:13]: 这一块儿搞完大概就没味道了，我换一些其他的正常的词吧😁

Comment (baiwusanyu-c) [reply docs/example/badge/max-value.svelte:13]: 其实玩梗没问题，我只想藏的深一点🤣

File: preset/src/shortcuts/badge.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+export const badgeShortcuts: Record<string, string> = {
+	// badge
+	'k-badge--base': 'relative v-mid inline-block w-fit',
```
Comment (baiwusanyu-c) [inline preset/src/shortcuts/badge.ts:3]:
```suggestion
	'k-badge--base': 'pr v-mid inline-block w-fit',
```

File: docs/components/KBadge.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+:::
+
+## Basic usage
+
+Displays the amount of new messages.
+
+The amount is defined with value which accepts Number or String.
+
+<demo src="../../../../example/badge/basic.svelte"  github='https://github.com/ikun-svelte/ikun-ui/tree/main/components/Badge'></demo>
```
Comment (baiwusanyu-c) [inline docs/components/KBadge.md:34]:
```suggestion
<demo src="../../../../example/badge/basic.svelte"  github='Badge'></demo>
```
Just pass the component name

**Review by baiwusanyu-c — CHANGES_REQUESTED** [review]

**Review by baiwusanyu-c — APPROVED** [review]

**Pull request merged**